src/logger.ts

```diff
@@ -65,6 +69,44 @@ export class Logger {
 
 		this.backend.unmanaged = unmanaged;
 		this.backend.publishEnabled = enableLogs;
+
+		// Only setup a config listener if we have to
+		if (!this.balenaBackend.initialised) {
```

Comment:
I don't understand this logic fully yet, but should this flag be swapped in the result of the actions below?
I don't see any code like that.

Reply:
Ah, found it in `assignFields`...

Reply:
@CameronDiver Want to give some thoughts on the matter, but I'm fully ok with merging this for now.

I may lack some extra background, but based on my current experience/impressions.

`balenaBackend` seems to be initialized asynchronously.

If we want the users of balena backend to deal with that, my suggestion would to work with the backend as a promise, so that for the user it would look like.

```
const client = await this.balenaBackend();
client.callSomeApi();
```

I don't really like this approach, but the main point is that it's a bit weird that user of `balenaBackend` has to deal with its initialisation. There may be many users - we get a question who then is responsible for full initialization. Ultimately, it results in a situation when every user has to care about this.

A more prefereable approach would be to hide asynchronous initalization in the backend class.
API calls are asynchronous anyway, so if we hide asynchronous initalization behind that API call promise, the user will not feel the difference. Like
```
// user code:
const res = await this.balenaBackend.callSomeApi();

// BalenaBackend class:
public async callSomeApi() {
  await config();
  return await doCallSomeApi();
}
```

To summarize: state has to be encapsulated and hidden in balenaBackend. Users of balenaBackend, should not care about its `initalized` state. Hope it makes sense.